paasta_tools/utils.py

```diff
@@ -1015,6 +1015,11 @@ def get_deploy_whitelist(self):
         """
         return self.get("deploy_whitelist", [])
 
+    def get_expected_slave_attributes(self):
+        """Return a list of dictionaries, representing the expected combinations of attributes in this cluster. Used for
```

Comment:
why's this a list of dictionaries, rather than just a dictionary of ``{ discover_type: possible_values}``?